### PR TITLE
Draft of backup/restore overview

### PR DESCRIPTION
**Pull request by kathancox**
Fixes [DOC-2458](https://cockroachlabs.atlassian.net/browse/DOC-2458)

This PR adds a Backup and Restore Overview page at the top of the BU/Restore navigation. The intention for this page is provide a landing area for users to understand bu/restore types, product availability, and useful features. Moving forward as significant features are added to both operations this gives us a space to direct our users through the doc set.

### CODE REVIEW

**Comment by kathancox** [comment]
Preview link: https://deploy-preview-13956--cockroachdb-docs.netlify.app/docs/stable/backup-and-restore-overview.html

File: v22.1/backup-and-restore-overview.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+| [`ALTER BACKUP`](alter-backup.html) | Add a new [KMS encryption key](take-and-restore-encrypted-backups.html#use-key-management-service) to an encrypted backup.                                        |
+| [`BACKUP`](backup.html)       | Create full and incremental backups.                                                        |
+| [`CANCEL JOB`](cancel-job.html)   | Use with the `job ID` to cancel a backup or restore job.                                    |
+| [`PAUSE JOB`](pause-job.html)    | Use with the `job ID` to pause a backup or restore job.                                     |
+| [`RESTORE`](restore.html)      | Restore full and incremental backups.                                                       |
+| [`RESUME JOB`](resume-job.html)   | Use with the `job ID` to resume a backup or restore job.                                    |
+| [`SHOW BACKUP`](show-backup.html)  | Show a backup's details at the [backup collection's](take-full-and-incremental-backups.html#backup-collections) storage location.                        |
+| [`SHOW JOBS`](show-jobs.html)    | Use to show a list of jobs or with the `job ID` to show details of a backup or restore job. |
+
```
Comment (livlobo) [inline v22.1/backup-and-restore-overview.md:45]: These are alphabetically ordered here, but would it make sense to order them in a the order of usage patterns?
e.g. 

BACKUP - start a backup
SHOW JOBS - show jobs to see status of my backup job
PAUSE/RESUME/CANCEL JOB -- pause/resume' cancel my backup job
SHOW BACKUP -- show details of a complete backup
RESTORE -- restore from a completed backup
ALTER BACKUP -- if kms rotated keys, add new keys to existing chain

**Review by livlobo — APPROVED** [review]
LGTM

**Review by andf-crl — APPROVED** [review]
Excellent! This really helps bring all the moving pieces together, and presents a highly useful at-a-glance view of supported features of CRDB's B&R capabilities. Nice work! I have penned a few minor nits regarding phrasing and structure, none are blocking. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13956#-:-N4Tfuqs9arqOQslcfvw:b-x6nxim)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @kathancox, and @livlobo)

___
*[`v22.1/backup-and-restore-overview.md` line 12 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TXvA88KHpjPW4s4H-:-N4TXvA88KHpjPW4s4H0:bimeku4) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L12)):*
> ```Markdown
> - The [types of backup and restore](#backup-and-restore-types) and their product availability
> - [Storage](#backup-storage) for backups
> - [SQL statements](#backup-and-restore-sql-statements) for working with backups and restores
> ```

nit: ideally, the order of the ToC should match the order of headings below (i.e. `storage` and `SQL statements` are switched below)
___
*[`v22.1/backup-and-restore-overview.md` line 14 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TY8d9ELvTeTPHXeys:-N4TY8d9ELvTeTPHXeyt:bsg1x1e) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L14)):*
> ```Markdown
> - [SQL statements](#backup-and-restore-sql-statements) for working with backups and restores
> 
> You can create full or incremental backups of a [cluster](backup.html#backup-a-cluster), [database](backup.html#backup-a-database), or [table](backup.html#backup-a-table-or-view). Taking regular backups of your data is an operational best practice.
> ```

Love this sentence pair: full use case scope (2: full + incremental && 3: cluster + database + table) and prominent production guidance. Perfect place for it also. Nice!
___
*[`v22.1/backup-and-restore-overview.md` line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TYU66DtMvmt208r6j:-N4TYU66DtMvmt208r6k:b-coc9hc) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L16)):*
> ```Markdown
> You can create full or incremental backups of a [cluster](backup.html#backup-a-cluster), [database](backup.html#backup-a-database), or [table](backup.html#backup-a-table-or-view). Taking regular backups of your data is an operational best practice.
> 
> ## Backup and restore types
> ```

nit: ideally this title matches its corresponding ToC entry, but these are inverted from one another. Minor indeed.
___
*[`v22.1/backup-and-restore-overview.md` line 29 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TZ2aH-RfY2I0tlXGV:-N4TZ2aH-RfY2I0tlXGW:b6ji4xi) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L29)):*
> ```Markdown
> | [Locality-aware backup and restore](take-and-restore-locality-aware-backups.html) | A backup where each node writes files only to the backup destination that matches the node locality configured at node startup.                              | {{ site.data.products.dedicated }} or with an [{{ site.data.products.enterprise }} license](enterprise-licensing.html) in CockroachDB.                                                                                                                                       |
> | [Scheduled backup](manage-a-backup-schedule.html)                  | A schedule for periodic backups.                                                                                                                             | {{ site.data.products.dedicated }} or with an [{{ site.data.products.enterprise }} license](enterprise-licensing.html) in CockroachDB  
> | [Automated Backups](../cockroachcloud/backups-page.html)                 | Cockroach Labs runs full backups daily and incremental backups hourly for every CockroachDB Cloud cluster.                                                   | {{ site.data.products.db }} clusters                                                                                                                                     
> ```

nit: `Automated Backups` should match title case with the rest of entries. Similarly, it should also be singular to match the rest of entries.

nit: `CockroachDB Dedicated or with an Enterprise license in CockroachDB` could be clarified as `CockroachDB Dedicated<br>CockroachDB Self-Hosted with an [Enterprise license]`. This helps to align with text `Self-Hosted` already present in left-nav (which is otherwise not present on this overview page), and helps to differentiate that Enterprise has nothing to do with Dedicated: `with` somewhat infers.
___
*[`v22.1/backup-and-restore-overview.md` line 44 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TcfvaD7OHm6-39Vbq:-N4TcfvaD7OHm6-39Vbr:b-o74j7j) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L44)):*
> ```Markdown
> | [`RESUME JOB`](resume-job.html)   | Use with the `job ID` to resume a backup or restore job.                                    |
> | [`SHOW BACKUP`](show-backup.html)  | Show a backup's details at the [backup collection's](take-full-and-incremental-backups.html#backup-collections) storage location.                        |
> | [`SHOW JOBS`](show-jobs.html)    | Use to show a list of jobs or with the `job ID` to show details of a backup or restore job. |
> ```

nit: mentions of `job ID` should ideally be moved to the end, such that the first words of every entry in column 2 of this table answer the question "what do I use column 1 for?". Example: `Cancel a backup or restore job by its job ID`.

nit:`SHOW JOBS` could be simplified with `Show a list of all running jobs or show the details of a specific job by its job ID.`
___
*[`v22.1/backup-and-restore-overview.md` line 45 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N305uE-u3dw1pMNhUTn-r1-45:-N4TcBI-DjvV6x8RoKuD:b-w4c4p5) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L45)):*
<details><summary><i>Previously, livlobo wrote…</i></summary><blockquote>

These are alphabetically ordered here, but would it make sense to order them in a the order of usage patterns?
e.g. 

BACKUP - start a backup
SHOW JOBS - show jobs to see status of my backup job
PAUSE/RESUME/CANCEL JOB -- pause/resume' cancel my backup job
SHOW BACKUP -- show details of a complete backup
RESTORE -- restore from a completed backup
ALTER BACKUP -- if kms rotated keys, add new keys to existing chain
</blockquote></details>

This is a tough one. I would advise simply matching the approach used elsewhere (alpha or usage) in the corpus. But this is not standard throughout, so no standard can be considered here. I support either choice made by the two of you.
___
*[`v22.1/backup-and-restore-overview.md` line 51 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4Te0Wm4v5hYc7EDS0p:-N4Te0Wm4v5hYc7EDS0q:b6e5bri) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L51)):*
nit: `set up` is reused for two nearly-identical concepts: `authentication` and `credentials`, which makes this read a bit awkwardly. What about something like:

> To set up authentication to a cloud storage bucket and include those credentials in the URL, read [Authentication](use-cloud-storage-for-bulk-operations.html#authentication).
___
*[`v22.1/backup-and-restore-overview.md` line 56 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TfFJG7B-x4YB0CEQt:-N4TfFJG7B-x4YB0CEQu:b3inh6t) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L56)):*
> ```Markdown
> 
> - [Object locking](use-cloud-storage-for-bulk-operations.html#object-locking) to prevent backups from being overwritten or deleted.
> - [Storage Class (AWS S3 only)](use-cloud-storage-for-bulk-operations.html#amazon-s3-storage-classes) to set a specific storage class for your backups.
> ```

nit: ideally these two bullet points are presented in parallel fashion to the previous two: `To prevent backups from being overwritten or deleted, read [Object locking]`. For my part, I prefer this second approach. It also matches the seealso approach immediately following. However, the full-sentence approach in the earlier two bullet points do read more clearly. Up to you; I just prefer that the approach is uniform.

**Comment by kathancox** [comment]
> Excellent! This really helps bring all the moving pieces together, and presents a highly useful at-a-glance view of supported features of CRDB's B&R capabilities. Nice work! I have penned a few minor nits regarding phrasing and structure, none are blocking.
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13956#-:-N4Tfuqs9arqOQslcfvw:b-x6nxim)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andf-crl, @kathancox, and @livlobo)
> 
> _[`v22.1/backup-and-restore-overview.md` line 12 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TXvA88KHpjPW4s4H-:-N4TXvA88KHpjPW4s4H0:bimeku4) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L12)):_
> 
> > ```gfm
> > - The [types of backup and restore](#backup-and-restore-types) and their product availability
> > - [Storage](#backup-storage) for backups
> > - [SQL statements](#backup-and-restore-sql-statements) for working with backups and restores
> > ```
> 
> nit: ideally, the order of the ToC should match the order of headings below (i.e. `storage` and `SQL statements` are switched below)
> 
> _[`v22.1/backup-and-restore-overview.md` line 14 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TY8d9ELvTeTPHXeys:-N4TY8d9ELvTeTPHXeyt:bsg1x1e) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L14)):_
> 
> > ```gfm
> > - [SQL statements](#backup-and-restore-sql-statements) for working with backups and restores
> > 
> > You can create full or incremental backups of a [cluster](backup.html#backup-a-cluster), [database](backup.html#backup-a-database), or [table](backup.html#backup-a-table-or-view). Taking regular backups of your data is an operational best practice.
> > ```
> 
> Love this sentence pair: full use case scope (2: full + incremental && 3: cluster + database + table) and prominent production guidance. Perfect place for it also. Nice!
> 
> _[`v22.1/backup-and-restore-overview.md` line 16 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TYU66DtMvmt208r6j:-N4TYU66DtMvmt208r6k:b-coc9hc) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L16)):_
> 
> > ```gfm
> > You can create full or incremental backups of a [cluster](backup.html#backup-a-cluster), [database](backup.html#backup-a-database), or [table](backup.html#backup-a-table-or-view). Taking regular backups of your data is an operational best practice.
> > 
> > ## Backup and restore types
> > ```
> 
> nit: ideally this title matches its corresponding ToC entry, but these are inverted from one another. Minor indeed.
> 
> _[`v22.1/backup-and-restore-overview.md` line 29 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TZ2aH-RfY2I0tlXGV:-N4TZ2aH-RfY2I0tlXGW:b6ji4xi) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L29)):_
> 
> > ```gfm
> > | [Locality-aware backup and restore](take-and-restore-locality-aware-backups.html) | A backup where each node writes files only to the backup destination that matches the node locality configured at node startup.                              | {{ site.data.products.dedicated }} or with an [{{ site.data.products.enterprise }} license](enterprise-licensing.html) in CockroachDB.                                                                                                                                       |
> > | [Scheduled backup](manage-a-backup-schedule.html)                  | A schedule for periodic backups.                                                                                                                             | {{ site.data.products.dedicated }} or with an [{{ site.data.products.enterprise }} license](enterprise-licensing.html) in CockroachDB  
> > | [Automated Backups](../cockroachcloud/backups-page.html)                 | Cockroach Labs runs full backups daily and incremental backups hourly for every CockroachDB Cloud cluster.                                                   | {{ site.data.products.db }} clusters                                                                                                                                     
> > ```
> 
> nit: `Automated Backups` should match title case with the rest of entries. Similarly, it should also be singular to match the rest of entries.
> 
> nit: `CockroachDB Dedicated or with an Enterprise license in CockroachDB` could be clarified as `CockroachDB Dedicated<br>CockroachDB Self-Hosted with an [Enterprise license]`. This helps to align with text `Self-Hosted` already present in left-nav (which is otherwise not present on this overview page), and helps to differentiate that Enterprise has nothing to do with Dedicated: `with` somewhat infers.
> 
> _[`v22.1/backup-and-restore-overview.md` line 44 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TcfvaD7OHm6-39Vbq:-N4TcfvaD7OHm6-39Vbr:b-o74j7j) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L44)):_
> 
> > ```gfm
> > | [`RESUME JOB`](resume-job.html)   | Use with the `job ID` to resume a backup or restore job.                                    |
> > | [`SHOW BACKUP`](show-backup.html)  | Show a backup's details at the [backup collection's](take-full-and-incremental-backups.html#backup-collections) storage location.                        |
> > | [`SHOW JOBS`](show-jobs.html)    | Use to show a list of jobs or with the `job ID` to show details of a backup or restore job. |
> > ```
> 
> nit: mentions of `job ID` should ideally be moved to the end, such that the first words of every entry in column 2 of this table answer the question "what do I use column 1 for?". Example: `Cancel a backup or restore job by its job ID`.
> 
> nit:`SHOW JOBS` could be simplified with `Show a list of all running jobs or show the details of a specific job by its job ID.`
> 
> _[`v22.1/backup-and-restore-overview.md` line 45 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N305uE-u3dw1pMNhUTn-r1-45:-N4TcBI-DjvV6x8RoKuD:b-w4c4p5) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L45)):_
> 
> Previously, livlobo wrote…
> This is a tough one. I would advise simply matching the approach used elsewhere (alpha or usage) in the corpus. But this is not standard throughout, so no standard can be considered here. I support either choice made by the two of you.
> 
> _[`v22.1/backup-and-restore-overview.md` line 51 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4Te0Wm4v5hYc7EDS0p:-N4Te0Wm4v5hYc7EDS0q:b6e5bri) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L51)):_ nit: `set up` is reused for two nearly-identical concepts: `authentication` and `credentials`, which makes this read a bit awkwardly. What about something like:
> 
> > To set up authentication to a cloud storage bucket and include those credentials in the URL, read [Authentication](use-cloud-storage-for-bulk-operations.html#authentication).
> 
> _[`v22.1/backup-and-restore-overview.md` line 56 at r1](https://reviewable.io/reviews/cockroachdb/docs/13956#-N4TfFJG7B-x4YB0CEQt:-N4TfFJG7B-x4YB0CEQu:b3inh6t) ([raw file](https://github.com/cockroachdb/docs/blob/c5e29140763d97a4c8618586ff4a11ba286a33be/v22.1/backup-and-restore-overview.md#L56)):_
> 
> > ```gfm
> > - [Object locking](use-cloud-storage-for-bulk-operations.html#object-locking) to prevent backups from being overwritten or deleted.
> > - [Storage Class (AWS S3 only)](use-cloud-storage-for-bulk-operations.html#amazon-s3-storage-classes) to set a specific storage class for your backups.
> > ```
> 
> nit: ideally these two bullet points are presented in parallel fashion to the previous two: `To prevent backups from being overwritten or deleted, read [Object locking]`. For my part, I prefer this second approach. It also matches the seealso approach immediately following. However, the full-sentence approach in the earlier two bullet points do read more clearly. Up to you; I just prefer that the approach is uniform.

TFTR @andf-crl I have implemented all your feedback! And I have gone with Liv's approach for now re the sql statements. We can adapt in the future if we feel differently 👍

**Pull request merged**